lua/wikis/mobilelegends/Infobox/Character/Custom.lua

```diff
@@ -0,0 +1,214 @@
+		} or nil,
+		String.isNotEmpty(args.costlg) and HtmlWidgets.Fragment{
+			children = {CostDisplay.display('lucky gem', '15px', args.costlg)}
+		} or nil,
+		String.isNotEmpty(args.costticket) and HtmlWidgets.Fragment{
+			children = {CostDisplay.display('ticket', '15px', args.costticket)}
+		} or nil
+	)
+	return Cell{name = 'Price', content = costContent}
+end
+
+---@return Widget[]
+function CustomCharacter:_getCustomCells()
+	local args = self.args
+	local widgets = {}
+	Array.appendWith(
+		widgets,
+		Cell{name = 'Specialty', content = self:_getCharacterSpecialties()},
+		Cell{name = 'Attack Type', content = {args.attacktype}},
+		Cell{name = 'Resource Bar', content = {args.resourcebar}},
+		Cell{name = 'Voice Actor(s)', content = self:_getVoiceActors()}
+	)
+
+	local baseStats = {
+		{name = 'HP', value = args.hp},
+		{name = 'HP Regen', value = args.hpreg},
+		{name = 'Mana', value = args.mana},
+		{name = 'Mana Regen', value = args.manareg},
+		{name = 'Energy', value = args.energy},
+		{name = 'Energy Regen', value = args.energyreg},
+		{name = 'Physical Attack', value = args.phyatk},
+		{name = 'Physical Defense', value = args.phydef},
+		{name = 'Magic Power', value = args.mp},
+		{name = 'Magic Defense', value = args.magdef},
+		{name = 'Attack Speed', value = args.atkspeed},
+		{name = 'Attack Speed Ratio', value = args.atkspeedratio},
+		{name = 'Movement Speed', value = args.movespeed}
+	}
+
+	if Array.any(baseStats, function(item) return Logic.isNotEmpty(item.value) end) then
+		Array.appendWith(widgets, Title{children = 'Base Statistics'})
+	end
+
+	Array.extendWith(widgets, Array.map(baseStats, function(item)
+		return Cell{name = item.name, content = {item.value}}
+	end))
+
+	local wins, loses = CharacterWinLoss.run(args.name)
+	if wins + loses == 0 then return widgets end
+
+	local winPercentage = Math.round(wins * 100 / (wins + loses), 2)
+
+	return Array.append(widgets,
+		Title{children = 'Esports Statistics'},
+		Cell{name = 'Win Rate', content = {wins .. 'W : ' .. loses .. 'L (' .. winPercentage .. '%)'}}
+	)
+end
+
+---@return string[]
+function CustomCharacter:_getCharacterSpecialties()
+	local args = self.args
+	local specialties = {}
+	for _, specialty in Table.iter.pairsByPrefix(args, 'specialty', {requireIndex = false}) do
+		table.insert(specialties, specialty)
+	end
+	return specialties
+end
+
+---@return string[]
+function CustomCharacter:_getVoiceActors()
+	local args = self.args
+	local voiceActors = {}
+	for voiceActorKey, voiceActor in Table.iter.pairsByPrefix(args, 'voice', {requireIndex = false}) do
+		local flag = args[voiceActorKey .. 'flag']
+		if flag then
+			voiceActor = Flags.Icon{flag = flag} .. ' ' .. voiceActor
+		end
+		table.insert(voiceActors, voiceActor)
+	end
+	return voiceActors
+end
+
+---@param args table
+---@return string[]
+function CustomCharacter:getWikiCategories(args)
+	if not Namespace.isMain() then return {} end
+	return WidgetUtil.collect(
```

Comment:
```suggestion
	return Array.append({},
```
imho only use widget util in widget contexts not for collecting category strings